src/progress_bar.rs

```diff
@@ -187,7 +187,7 @@ impl ProgressBar {
     pub fn set_draw_rate(&self, n: u64) {
         let mut state = self.state.lock().unwrap();
         state.draw_rate = n;
-        state.draw_next = state.pos.saturating_add(state.per_sec() / n);
+        state.draw_next = state.pos.saturating_add(state.per_sec() as u64 / n);
```

Comment:
I think we would get better data if the conversion to `u64` wraps the division, right?

Reply:
Fixed. I cast both values to `f64` now and then cast to `u64` at the end.